scripts/named_entity_recognition/code/train_model.py

```diff
@@ -0,0 +1,379 @@
+    for sent in sentences:
+        per_chars = []
+        for word in sent:
+            per_chars.append(pad_chars(char_vocab[list(word)]))
+        per_chars_of_sentences.append(per_chars)
+    sentences = [word_vocab[sent] for sent in sentences]
+    tags = [tag_vocab[tags] for tags in sentences_tags]
+
+    # dataset [[x,x,x], [[c,c], [c,c], [c,c]], [t, t, t]]
+    dataset = gluon.data.SimpleDataset([[sent, per_chars, tag] for sent, per_chars,
+                                        tag in zip(sentences, per_chars_of_sentences, tags) if len(sent) > 0])
+
+    return dataset
+
+
+def data_pipeline(train_data_path, valid_data_path, test_data_path,
+                  embedding, word_vocab_path, char_vocab_path,
+                  tag_vocab_path, batch_size, max_char_len, logger):
+    '''
+    Get the data and build a data pipeline to prepare for the training model.
+
+    Args:
+        train_data_path (str): path of train data
+        valid_data_path (str): path of valid data
+        test_data_path (str): path of test data
+        embedding (str): the category of pre-trained embedding, ['glove', 'senna', 'random']
+        word_vocab_path (str): path of word vocab
+        char_vocab_path (str): path of char vocab
+        tag_vocab_path (str): path of tag vocab
+        batch_size (int): batch size
+        max_char_len (int): The maximum number of characters each word has
+        logger (logger): logging some data
+
+    Returns:
+        char_vocab (nlp.Vocab) : Vocab of characters
+        word_vocab (nlp.Vocab) : Vocab of words
+        tag_vocab (nlp.Vocab) : Vocab of tags
+        train_dataloader (gluon.data.DataLoader) : dataloader of train data
+        valid_dataloader (gluon.data.DataLoader) : dataloader of valid data
+        test_dataloader (gluon.data.DataLoader) : dataloader of test data
+        [type]: [description]
+    '''
+
+    # get data and transform to bio2 format
+    train_sentences, train_sentences_tags = get_data_bio2(train_data_path)
+    valid_sentences, valid_sentences_tags = get_data_bio2(valid_data_path)
+    test_sentences, test_sentences_tags = get_data_bio2(test_data_path)
+
+    logger.info('train_sentences length: %d, valid_sentences length: %d, test_sentences length: %d'
+                % (len(train_sentences), len(valid_sentences), len(test_sentences)))
+
+    # build word_vocab, char_vocab, tag_vocab
+    all_sentences = train_sentences + valid_sentences + test_sentences
+    all_tags = train_sentences_tags + valid_sentences_tags + test_sentences_tags
+    counter = nlp.data.count_tokens(list(itertools.chain.from_iterable(all_sentences)))
+    word_vocab = nlp.Vocab(counter, max_size=50000)
+
+    char_counter = nlp.data.count_tokens(list(''.join([''.join(sentence) for sentence in all_sentences])))
+    char_vocab = nlp.Vocab(char_counter)
+
+    tag_counter = nlp.data.count_tokens(list(itertools.chain.from_iterable(all_tags)))
+    tag_vocab = nlp.Vocab(tag_counter, padding_token='O')
```

Comment:
```suggestion
    tag_vocab = nlp.Vocab(tag_counter, padding_token='O', bos_token=None, eos_token=None, unknown_token=None)
```
We do not need BOS/EOS/unknown tokens for tags.